orchagent/vxlanorch.cpp

```diff
@@ -515,30 +641,532 @@ bool VxlanTunnel::removeNextHop(IpAddress& ipAddr, MacAddress macAddress, uint32
+           {
+               ids_.tunnel_decap_id[TUNNEL_MAP_T_VIRTUAL_ROUTER] = vtep_ptr->getDecapMapId(TUNNEL_MAP_T_VIRTUAL_ROUTER);
+               ids_.tunnel_encap_id[TUNNEL_MAP_T_VIRTUAL_ROUTER] = create_tunnel_map(MAP_T::VRID_TO_VNI);
+               TUNNELMAP_SET_VRF(encap_dedicated_mappers_);
+           }
+
+           if (IS_TUNNELMAP_SET_BRIDGE(mapper_list))
+           {
+               ids_.tunnel_decap_id[TUNNEL_MAP_T_BRIDGE] = vtep_ptr->getDecapMapId(TUNNEL_MAP_T_BRIDGE);
+               ids_.tunnel_encap_id[TUNNEL_MAP_T_BRIDGE] = create_tunnel_map(MAP_T::VRID_TO_VNI);
+               TUNNELMAP_SET_BRIDGE(encap_dedicated_mappers_);
+           }
+
+        }
+        else if (TUNNEL_MAP_USE_COMMON_ENCAP_DECAP == map_src)
+        {
+           if (IS_TUNNELMAP_SET_VLAN(mapper_list))
+           {
+             ids_.tunnel_decap_id[TUNNEL_MAP_T_VLAN] = vtep_ptr->getDecapMapId(TUNNEL_MAP_T_VLAN);
+             ids_.tunnel_encap_id[TUNNEL_MAP_T_VLAN] = vtep_ptr->getEncapMapId(TUNNEL_MAP_T_VLAN);
+           }
+
+           if (IS_TUNNELMAP_SET_VRF(mapper_list))
+           {
+             ids_.tunnel_decap_id[TUNNEL_MAP_T_VIRTUAL_ROUTER] = vtep_ptr->getDecapMapId(TUNNEL_MAP_T_VIRTUAL_ROUTER);
+             ids_.tunnel_encap_id[TUNNEL_MAP_T_VIRTUAL_ROUTER] = vtep_ptr->getEncapMapId(TUNNEL_MAP_T_VIRTUAL_ROUTER);
+           }
+
+           if (IS_TUNNELMAP_SET_BRIDGE(mapper_list))
+           {
+             ids_.tunnel_decap_id[TUNNEL_MAP_T_BRIDGE] = vtep_ptr->getDecapMapId(TUNNEL_MAP_T_BRIDGE);
+             ids_.tunnel_encap_id[TUNNEL_MAP_T_BRIDGE] = vtep_ptr->getEncapMapId(TUNNEL_MAP_T_BRIDGE);
+           }
+
+        }
+        else if (TUNNEL_MAP_USE_DECAP_ONLY == map_src)
+        {
+           if (IS_TUNNELMAP_SET_VLAN(mapper_list))
+           {
+             ids_.tunnel_decap_id[TUNNEL_MAP_T_VLAN] = create_tunnel_map(MAP_T::VNI_TO_VLAN_ID);
+             TUNNELMAP_SET_VLAN(decap_dedicated_mappers_);
+           }
+
+           if (IS_TUNNELMAP_SET_VRF(mapper_list))
+           {
+             ids_.tunnel_decap_id[TUNNEL_MAP_T_VIRTUAL_ROUTER] = create_tunnel_map(MAP_T::VNI_TO_VRID);
+             TUNNELMAP_SET_VRF(decap_dedicated_mappers_);
+           }
+
+           if (IS_TUNNELMAP_SET_BRIDGE(mapper_list))
+           {
+             ids_.tunnel_encap_id[TUNNEL_MAP_T_BRIDGE] = create_tunnel_map(MAP_T::BRIDGE_TO_VNI);
+
+             TUNNELMAP_SET_BRIDGE(decap_dedicated_mappers_);
+           }
+        }
+    }
+
+    catch (const std::runtime_error& error)
+    {
+        SWSS_LOG_ERROR("Error creating tunnel %s: %s", tunnel_name_.c_str(), error.what());
+        return false;
+    }
+
+    return true;
+}
+
+bool VxlanTunnel::deleteTunnelHw(uint8_t mapper_list, 
+                                 tunnel_map_use_t map_src, bool with_term)
+{
+    try
+    {
+      if (with_term)
+      {
+          remove_tunnel_termination(ids_.tunnel_term_id);
+      }
+
+      remove_tunnel_termination(ids_.tunnel_id);
```

Comment:
In case of tunnel p2p mode, one tunnel mapper entry could be part of multiple tunnel objects. In that case, I think it is better to check any valid tunnelObject references before removing the mapper entry.

Reply:
p2p tunnels would have used TUNNEL_MAP_USE_COMMON_ENCAP_DECAP which is a no-op in the deleteMapperHw. So this should not be an issue